arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java

```diff
@@ -115,23 +110,28 @@ public SelfdestructSection(Hub hub) {
     if (Exceptions.any(exceptions)) {
       checkArgument(exceptions == OUT_OF_GAS_EXCEPTION);
 
-      selfDestroyerFirstAccountFragment =
+      recipientAccountBefore =
+          selfdestructTargetsItself
+              ? selfdestructorAccountBefore
+              : AccountSnapshot.canonical(hub, recipientAddress);
+
+      selfdestructorFirstAccountFragment =
```

Comment:
In case of an OOGX there is only `Before` data. We adjst the snapshots accordingly.